src/tests/ftest/util/general_utils.py

```diff
@@ -304,3 +306,25 @@ def verify_rebuild(pool, log, to_be_rebuilt, object_qty, record_qty, errors=0):
                 "Unexpected {} value: expected={}, detected={}".format(
                     key, expected, detected))
     return messages
+
+def clustershell_execute(cmd, hosts, timeout=60):
+    """
+    Run command on given hosts using cluster shell utility.
+    Args:
+        cmd (str): command in string format to run on cluster
+        hosts (list):  a list of host names
+        timeout (int): command timeout in seconds
+    Raise:
+        ClusterCommandFailed: raied in case of command failed on any host
+                              return dictionary with hostname,rc,output.
+    """
+    rc_err = {}
+    local_task = task_self()
+    local_task.run(cmd, nodes=','.join(hosts), timeout=timeout)
+    for rc_code, _keys in local_task.iter_retcodes():
+        if rc_code is not 0:
+            rc_err[_keys[0]] = 'RC={}, output={}'.format(rc_code,
+                                                         local_task.
+                                                         node_buffer(_keys[0]))
+    if rc_err:
+        raise ClusterCommandFailed(rc_err)
```

Comment:
This could possibly be a future enhancement.  If we consider a worse case scenario where we have 100's of nodes executing the same command, it would be really nice to provide an error message that looked something like this, instead of dumping a dictionary containing many keys.

```
Error issuing 'some_command':
- node[1-15,37,78-83]: FAILED, rc=12, output=Write access error
- node[94]: FAILED, rc=1, output=Unable to connect
- node[33]: TIMEOUT, rc=133, output=Waiting....
```

Modifying rc_err such that keys are either the return codes and values are a list of the failure type, a `ClusterShell.NodeSet` object, and the error message would allow you to do the following:

`raise ClusterCommandFailed("Error issuing {}:\n{}".format(cmd, "\n".join(["- {}: {}, rc={}, output={}".format(rc_err[rc][0], rc_err[rc][1], rc, rc_err[rc][2]) for rc in rc_err])))`

A better approach would probably make use of the event handler and log the failed/timedout commands as they occur and then just raise an error message with just the nodes that were unable to successfully issue the command.